src/main/java/uk/gov/hmcts/ccd/config/TracingConfiguration.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.hmcts.ccd.config;
+
+import com.microsoft.applicationinsights.extensibility.TelemetryProcessor;
+import com.microsoft.applicationinsights.telemetry.RemoteDependencyTelemetry;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import uk.gov.hmcts.ccd.appinsights.CallbackTelemetryContext;
+import uk.gov.hmcts.ccd.appinsights.CallbackTelemetryThreadContext;
+
+@Configuration
+public class TracingConfiguration {
+
+    @Bean
+    // Custom TelemetryProcessor which tags the type of {@link RemoteDependencyTelemetry} as callback before publishing.
+    public TelemetryProcessor callbackRecognitionProcessor() {
+        return telemetry -> {
+            if (telemetry instanceof RemoteDependencyTelemetry) {
+                RemoteDependencyTelemetry dependency = (RemoteDependencyTelemetry) telemetry;
+                if (dependency.getType().startsWith("Http")
+                    && CallbackTelemetryThreadContext.getTelemetryContext() != null) {
+
+                    CallbackTelemetryContext telemetryContext = CallbackTelemetryThreadContext.getTelemetryContext();
+                    dependency.getProperties().put("callback", "true");
```

Comment:
with this solution we assume if a telemetry context exists then the dependency is a callback. We could make this a bit more generic and reusable for non-callback dependencies by marking as callback if getCallbackType() is not blank?